client/src/pages/shop/Shop.jsx

```diff
@@ -36,6 +37,21 @@ const ShopPage = () => {
     location.pathname.includes("accessories") && setCategory(accessories);
   };
 
+  const selectSort = (options) => {
+    switch (options) {
+      case "a":
+        ascendingPrice();
+        break;
+      case "b":
+        decendingPrice();
+        break;
+      case "c":
+        latest();
+        break;
+      default:
+    }
+  };
+
```

Comment:
you have to pass the category and setCategory as arguments else they won't work.
```suggestion
  const selectSort = (options) => {
    switch (options) {
      case "a":
        ascendingPrice(category, setCategory);
        break;
      case "b":
        decendingPrice(category, setCategory);
        break;
      case "c":
        latest(category, setCategory);
        break;
      default:
    }
  };

```